src/Domain/SegmentId.cpp

```diff
@@ -0,0 +1,31 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/SegmentId.hpp"
+
+#include <boost/functional/hash.hpp>
+#include <limits>
+#include <ostream>
+
+SegmentId::SegmentId(const size_t refinement_level, const size_t index)
+    : refinement_level_(refinement_level), index_(index) {
+  ASSERT(index < two_to_the(refinement_level),
+         "index = " << index << ", refinement_level = " << refinement_level);
+}
+
+void SegmentId::pup(PUP::er& p) {
+  p | refinement_level_;
+  p | index_;
+}
+
+std::ostream& operator<<(std::ostream& os, const SegmentId& id) {
+  os << 'L' << id.refinement_level() << 'I' << id.index();
+  return os;
+}
+
```

Comment:
Wrap this function in `// LCOV_EXCL_START` and `// LCOV_EXCL_STOP` since we don't want to have to worry about it during testing.